### PR TITLE
Refactor ShareOnSocialMedia.js

### LINKED ISSUE #1691: Refactor ShareOnSocialMedia.js
**Actual Behaviour**

Make ShareOnSocialMedia consistent with codebase

**Expected Behaviour**

Refactor ShareOnSocialMedia.js

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #1691 
Changes: Following changes are made to make the component consistent with codebase:

- Ordered import statements
- Arrow functions instead of normal functions
- constant declared before components to make it consistent
- Removed unwanted state (styling of raised buttons)

Demo Link: https://pr-1691-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

File: src/components/ChatApp/Settings/ShareOnSocialMedia.js [hunk 1]
```diff
@@ -83,44 +92,35 @@ export default class ShareOnSocialMedia extends Component {
...
-      buttonDiv: {
-        marginTop: '10px',
-        marginBottom: '0px',
-        fontSize: '15px',
-        fontWeight: 'bold',
-      },
-    };
-
     return (
       <div style={this.props.divStyle}>
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/ShareOnSocialMedia.js]: Rename the prop to `containerStyle`

File: src/components/ChatApp/Settings/ShareOnSocialMedia.js [hunk 2]
```diff
@@ -83,44 +92,35 @@ export default class ShareOnSocialMedia extends Component {
...
         <div>
           <div>
-            <div style={styles.buttonDiv}>
+            <div style={style.buttonStyle}>
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/ShareOnSocialMedia.js]: Please deconstruct the styles object

File: src/components/ChatApp/Settings/ShareOnSocialMedia.js [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
-import Translate from '../../Translate/Translate.react';
+import PropTypes from 'prop-types';
 import FontIcon from 'material-ui/FontIcon';
 import RaisedButton from 'material-ui/RaisedButton';
+import Translate from '../../Translate/Translate.react';
 import urls from '../../../utils/urls';
-import PropTypes from 'prop-types';
 
-export default class ShareOnSocialMedia extends Component {
+const style = {
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/ShareOnSocialMedia.js]: Make it `styles`

File: src/components/ChatApp/Settings/ShareOnSocialMedia.js [hunk 4]
```diff
@@ -83,44 +92,35 @@ export default class ShareOnSocialMedia extends Component {
...
               <br />
               <RaisedButton
                 label={<Translate text="Share on Facebook" />}
-                style={this.state.style}
+                style={style.raisedButtonStyle}
                 backgroundColor="#3B5998"
                 labelColor="#fff"
                 icon={<FontIcon className="fa fa-facebook" />}
                 keyboardFocused={false}
                 onTouchTap={this.Share}
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/ShareOnSocialMedia.js]: Change the function name to `shareToFacebook`

File: src/components/ChatApp/Settings/ShareOnSocialMedia.js [hunk 5]
```diff
@@ -1,24 +1,33 @@
...
     this.state = {
-      style: {
-        margin: 20,
-        width: '220px',
-      },
       loginStatus: 'not connected',
     };
   }
 
   facebookLoginAPI = () => {
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/ShareOnSocialMedia.js]: Rename it to `initFacebookAPI`

**Comment by fragm3** [comment]
@akshatnitd made the changes.

File: src/components/ChatApp/Settings/ShareOnSocialMedia.js [hunk 6]
```diff
@@ -42,35 +51,10 @@ export default class ShareOnSocialMedia extends Component {
...
-  };
-
-  facebookLogin = () => {
-    window.FB.login(this.checkLoginStatus);
-  };
-
-  Share = () => {
+  shareToFacebook = () => {
     if (
       this.state.loginStatus === 'connected' ||
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/ShareOnSocialMedia.js]: The `loginStatus` is not changing, do we require this state?

Comment (fragm3) [reply src/components/ChatApp/Settings/ShareOnSocialMedia.js]: oh, now we can convert this to functional component. Great!

**Comment by fragm3** [comment]
@akshatnitd made the changes!

**Comment by fragm3** [comment]
Making new PR as this branch is outdated, having conflicts.

**Pull request closed without merging**